docker-compose.yml

```diff
@@ -0,0 +1,55 @@
+# Most of this docker compose file was copied from here: 
+# https://github.com/graphprotocol/graph-node/blob/5ba079fff124f621b6beeba9dc39b81920a2f85b/docker/docker-compose.yml
+
+version: '3.9'
+services:
+  graph-node:
+    image: graphprotocol/graph-node
+    ports:
+      - '8000:8000'
+      - '8001:8001'
+      - '8020:8020'
+      - '8030:8030'
+      - '8040:8040'
+    depends_on:
+      - ipfs
+      - postgres
+    extra_hosts:
+      - 172.20.0.1:host-gateway
+    env_file:
+      - .env
+    environment:
+      postgres_host: postgres
+      postgres_user: graph-node
+      postgres_pass: let-me-in
+      postgres_db: graph-node
+      ipfs: 'ipfs:5001'
+      ethereum: 'mainnet:${ETH_MAINNET_ALCHEMY_RPC} matic:${POLYGON_MAINNET_ALCHEMY_RPC} avalanche:${AVALANCHE_MAINNET_RPC}'
```

Comment:
So I've tried this:

1. Create `.env` with RPCs on `http://localhost:8545`
2. `anvil`
3. `docker compose up`

And the node can't connect to the `anvil` instance. Which makes sense because IIRC services running in compose can't "see" services on the host.

I know that Foundry/Anvil has a way to run it in docker as well: https://github.com/foundry-rs/book/issues/289

Maybe we can add add anvil services for these chains and automatically spin up the whole toolchain?

Reply:
Tried adding the following `anvil` service config to no avail:

```
  anvil:
    image: ghcr.io/foundry-rs/foundry:latest
    ports:
      - '8545:8545'
    entrypoint: 'anvil --fork-url https://rpc.ankr.com/eth --host 0.0.0.0'
```


```
subgraph-graph-node-1  | May 17 13:01:55.863 WARN Trying again after eth_getBlockByNumber(0, false) RPC call failed (attempt #12) with result Err(failed to send request: error sending request for url (http://127.0.0.1:8545/): error trying to connect: tcp connect error: Cannot assign requested address (os error 99)), provider: mainnet-rpc-0
```